src/Microsoft.Restier.EntityFramework/Model/ModelMapper.cs

```diff
@@ -60,10 +61,10 @@ public ModelMapper(Type dbContextType)
             if (property != null)
             {
                 var type = property.PropertyType;
-                if (type.IsGenericType &&
-                    type.GetGenericTypeDefinition() == typeof(DbSet<>))
+                var parent = GetGenericParent(type);
```

Comment:
Thanks for the trick; it is good to know the usage of such compile time dependencies. I pushed the changes.